main.go

```diff
@@ -115,8 +137,60 @@ func main() {
 	}
 
 	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+	if err := mgr.Start(ctx); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
+
+// Tee consumes the received channel and mirrors the messages into 2 new channels.
+func Tee[T any](ctx context.Context, in <-chan T) (_, _ <-chan T) {
+	out1, out2 := make(chan T), make(chan T)
+
+	OrDone := func(ctx context.Context, in <-chan T) <-chan T {
+		out := make(chan T)
+		go func() {
+			defer close(out)
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case i, ok := <-in:
+					if !ok {
+						return
+					}
+					select {
+					case out <- i:
+					case <-ctx.Done():
+					}
+				}
+			}
+		}()
+		return out
+	}
+
+	go func() {
+		defer close(out1)
+		defer close(out2)
+
+		for val := range OrDone(ctx, in) {
+			select {
+			case <-ctx.Done():
+
+			case out1 <- val:
+				select {
+				case <-ctx.Done():
+				case out2 <- val:
+				}
+
+			case out2 <- val:
+				select {
+				case <-ctx.Done():
+				case out1 <- val:
+				}
+			}
+		}
+	}()
+	return out1, out2
+}
```

Comment:
Question: was this fanout function something you put together just for this PR, or was it used from elsewhere? If it was taken from elsewhere are there some unit tests that could come along for the ride?

Reply:
This was taken from a library I've written, containing common concurrency patterns for channels (heavily inspired by Katherine Cox-Buday's book - Concurrency in Go).

I don't have unit tests for it, but I've been using it for a while so I felt comfortable adding it..
Let me know if unit tests are important for you here, I'll figure something out

Reply:
Ah yeah, the book with the shrew on it right? I have that one.

I did some experimenting locally and I do think we could probably approach this a few different ways, but in retrospect given the grand scheme of things we probably don't need to focus on this too much for this PR, it will be very easy to iterate on this as needed in follow-up PRs so we can keep this PR more focused on the task at hand.

Reply:
> Ah yeah, the book with the shrew on it right? I have that one.

Yes exactly! It's a nice one

> I did some experimenting locally and I do think we could probably approach this a few different ways, but in retrospect given the grand scheme of things we probably don't need to focus on this too much for this PR

Totally agree, it was just the fastest and simplest solution I could come up with (given that I had this functionality from before lol).

> It will be very easy to iterate on this as needed in follow-up PRs so we can keep this PR more focused on the task at hand.

👍